event-types.go

```diff
@@ -109,34 +209,34 @@ func newErrorEvent(b []byte) *VehEvent {
 }
 
 func newVehEvent(b []byte) (VehEvent, int) {
-	length := 10 + int(b[9])
+	length := 10 + int(b[ehLengthOffset.Start()])
 	return VehEvent{
 		DataFlag:  b[length] > 0x00,
-		Sequence:  binary.LittleEndian.Uint32(b[0:4]),
-		Timestamp: binary.LittleEndian.Uint32(b[4:8]),
-		EventType: int(b[8]),
+		Sequence:  binary.LittleEndian.Uint32(b[ehSeqNoOffset.Start():ehSeqNoOffset.End()]),
+		Timestamp: binary.LittleEndian.Uint32(b[ehTimestampOffset.Start():ehTimestampOffset.End()]),
+		EventType: int(b[ehEventTypeOffset.Start()]),
 	}, length
 }
 
 func newBootEvent(b []byte) *VehEvent {
 	eb, _ := newVehEvent(b)
 	eb.BootEvent = &VehBootEvent{
-		binary.LittleEndian.Uint32(b[10:14]),
-		fmt.Sprintf("%d.%d", b[14], b[15]),
-		uint32(b[16]),
-		uint32(b[17]),
+		binary.LittleEndian.Uint32(b[ebReasonOffset.Start():ebReasonOffset.End()]),
+		fmt.Sprintf("%d.%d", b[ebVersionOffset.Start()], b[ebVersionOffset.Start()+1]),
+		uint32(b[ebVersionOffset.Start()+2]),
+		uint32(b[ebVersionOffset.Start()+3]),
 	}
 	return &eb
 }
 
 func newSensorEvent(b []byte) *VehEvent {
 	eb, l := newVehEvent(b)
 	eb.SensorEvent = &VehSensorEvent{
-		float32(binary.LittleEndian.Uint16(b[10:12]) / 4),
-		float32(binary.LittleEndian.Uint16(b[12:14]) / 1000),
+		float32(binary.LittleEndian.Uint16(b[esTemperatureOffset.Start():esTemperatureOffset.End()]) / 4),
+		float32(binary.LittleEndian.Uint16(b[esBatteryOffset.Start():esBatteryOffset.End()]) / 1000),
 		make([]byte, l-14),
 	}
-	copy(eb.SensorEvent.Other, b[13:l-1])
+	copy(eb.SensorEvent.Other, b[esOtherOffset.Start()-1:l-1])
 	return &eb
```

Comment:
I am not sure WTF this line of code is doing, it does not look correct to me, but I have reproduce the behaviour in the new code. Can you please double check this?

Reply:
Its copying the contents of the slice into the Other field
What happens is that some unknown data (`"other": "X2orPw==",`) ends up getting written to the database